planner/cascades/transformation_rules.go

```diff
@@ -1781,3 +1783,123 @@ func isInnerJoinKeysContainUniqueKey(innerGroup *memo.Group, joinKeys *expressio
 	}
 	return false, nil
 }
+
+// TransformAggregateCaseToSelection convert Agg(case when) to Agg->Selection.
+type TransformAggregateCaseToSelection struct {
+	baseRule
+}
+
+// NewRuleTransformAggregateCaseToSelection creates a new Transformation TransformAggregateCaseToSelection.
+// The pattern of this rule is `Agg->X`.
+func NewRuleTransformAggregateCaseToSelection() Transformation {
+	rule := &TransformAggregateCaseToSelection{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandAggregation,
+		memo.EngineTiDBOnly,
+	)
+	return rule
+}
+
+// Match implements Transformation interface.
+func (r *TransformAggregateCaseToSelection) Match(expr *memo.ExprIter) bool {
+	agg := expr.GetExpr().ExprNode.(*plannercore.LogicalAggregation)
+	return len(agg.GetGroupByUsedCols()) == 0 && len(agg.AggFuncs) == 1 && len(agg.AggFuncs[0].Args) == 1 && r.isTwoOrThreeArgCase(agg.AggFuncs[0].Args[0])
```

Comment:
BTW, we'd better add the check `agg.IsCompleteModeAgg()` here?

Reply:
yes